Bottom: 546230af519e3539f1e7f13a89e7b1d050a64a76
Top:    e7cb574b8f6857c32e1cf2db70b19065f5d9efc1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 04:09:41 +0000

find.c: Kill "SetThing" with fire


---

diff --git a/src/find.c b/src/find.c
index d10499d..67e792c 100644
--- a/src/find.c
+++ b/src/find.c
@@ -257,7 +257,7 @@ typedef struct
  * some local identifiers
  */
 static Coord Bloat = 0;
-static void *thing_ptr1, *thing_ptr2, *thing_ptr3;
+//static void *thing_ptr1, *thing_ptr2, *thing_ptr3;
 static int thing_type;
 static bool User = false;    /* user action causing this */
 static bool drc = false;     /* whether to stop if finding something not found */
@@ -3396,6 +3396,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
     }
   /* now check the bloated condition */
   drc = false;
+#warning flag could be uninitialised here!
   ClearFlagOnAllObjects (false, flag);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
@@ -3480,10 +3481,8 @@ drc_callback (DataType *data, LayerType *layer, PolygonType *polygon,
   PinType *pin = (PinType *) ptr2;
   PadType *pad = (PadType *) ptr2;
 
-  thing_type = type;
-  thing_ptr1 = ptr1;
-  thing_ptr2 = ptr2;
-  thing_ptr3 = ptr2;
+  SetFubarThing (type, ptr1, ptr2, ptr2); /* Note deliberate repeat of ptr2 */
+
   switch (type)
     {
     case LINE_TYPE:
@@ -3672,7 +3671,7 @@ DRCAll (void)
             SET_FLAG (flag, line);
             DrawLine (layer, line);
             drcerr_count++;
-            SetThing (LINE_TYPE, layer, line, line);
+            SetFubarThing (LINE_TYPE, layer, line, line);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Line width is too thin"),
@@ -3716,7 +3715,7 @@ DRCAll (void)
             SET_FLAG (flag, arc);
             DrawArc (layer, arc);
             drcerr_count++;
-            SetThing (ARC_TYPE, layer, arc, arc);
+            SetFubarThing (ARC_TYPE, layer, arc, arc);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Arc width is too thin"),
@@ -3761,7 +3760,7 @@ DRCAll (void)
             SET_FLAG (flag, pin);
             DrawPin (pin);
             drcerr_count++;
-            SetThing (PIN_TYPE, element, pin, pin);
+            SetFubarThing (PIN_TYPE, element, pin, pin);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Pin annular ring too small"),
@@ -3793,7 +3792,7 @@ DRCAll (void)
             SET_FLAG (flag, pin);
             DrawPin (pin);
             drcerr_count++;
-            SetThing (PIN_TYPE, element, pin, pin);
+            SetFubarThing (PIN_TYPE, element, pin, pin);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Pin drill size is too small"),
@@ -3836,7 +3835,7 @@ DRCAll (void)
             SET_FLAG (flag, pad);
             DrawPad (pad);
             drcerr_count++;
-            SetThing (PAD_TYPE, element, pad, pad);
+            SetFubarThing (PAD_TYPE, element, pad, pad);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Pad is too thin"),
@@ -3881,7 +3880,7 @@ DRCAll (void)
             SET_FLAG (flag, via);
             DrawVia (via);
             drcerr_count++;
-            SetThing (VIA_TYPE, via, via, via);
+            SetFubarThing (VIA_TYPE, via, via, via);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Via annular ring too small"),
@@ -3913,7 +3912,7 @@ DRCAll (void)
             SET_FLAG (flag, via);
             DrawVia (via);
             drcerr_count++;
-            SetThing (VIA_TYPE, via, via, via);
+            SetFubarThing (VIA_TYPE, via, via, via);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Via drill size is too small"),
@@ -3958,7 +3957,7 @@ DRCAll (void)
             SET_FLAG (flag, line);
             DrawLine (layer, line);
             drcerr_count++;
-            SetThing (LINE_TYPE, layer, line, line);
+            SetFubarThing (LINE_TYPE, layer, line, line);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
             violation = pcb_drc_violation_new (_("Silk line is too thin"),
@@ -4010,7 +4009,7 @@ DRCAll (void)
             SET_FLAG (flag, element);
             DrawElement (element);
             drcerr_count++;
-            SetThing (ELEMENT_TYPE, element, element, element);
+            SetFubarThing (ELEMENT_TYPE, element, element, element);
             LocateError (&x, &y);
             BuildObjectList (&object_count, &object_id_list, &object_type_list);
